### PR TITLE
Less GitHub dependence

### PR DESCRIPTION
**Pull request by Hoverbear**
Furthers the cause of #107 and extracts out all the github-fetched data into `src/cli/mod.rs` for now, such that `push_new_release` and `ReleaseMetadata::build` are no longer dependent on the github related structs. This should make opening up other code forges easier.

There should be *no material code changes* here, only refactoring.

### CODE REVIEW

File: src/release_metadata.rs [hunk 1]
```diff
@@ -90,92 +87,63 @@ impl ReleaseMetadata {
...
         upload_name: String,
         mirror: bool,
         visibility: Visibility,
-        github_graphql_data_result: GithubGraphqlDataResult,
-        extra_labels: Vec<String>,
-        spdx_expression: Option<spdx::Expression>,
+        labels: Vec<String>,
+        spdx_identifier: Option<spdx::Expression>,
+        project_id: i64,
+        owner_id: i64,
```
Comment (colemickens) [inline src/release_metadata.rs:110]: what do we even do with these? Presumably we only trust the JWT derived ids on the server?

File: src/cli/mod.rs [hunk 2]
```diff
@@ -434,24 +457,63 @@ impl FlakeHubPushCli {
...
         };
 
+        let commit_count = match revision_info.local_revision_count {
+            Some(n) => n as i64,
+            None => {
+                tracing::debug!(
+                    "Getting revision count locally failed, using data from github instead"
+                );
+                github_graphql_data_result.rev_count
+            }
```
Comment (colemickens) [inline src/cli/mod.rs:467]: hmmm. not a blocker but can we just get rid of it completely? what scenarios does from_git_root fail, and does the rest of the push process still work even in that case?

**Review by colemickens — APPROVED** [review]
Looks good, don't want to get too carried away with ideas since I know there's more coming.

File: src/cli/mod.rs (hunk 2 above)
Comment (Hoverbear) [reply src/cli/mod.rs:467]: Generally github action clones are shallow if I recall? I'm not sure I remember. I'm hoping to get rid of querying this data entirely.

File: src/release_metadata.rs (hunk 1 above)
Comment (Hoverbear) [reply src/release_metadata.rs:110]: That's correct, I don't even see us deserializing these fields on the server, but I'm trying to keep this simple mechanical moves. Next PR is contentful. :)

**Comment by cole-h** [comment]
FYI: this was reverted in https://github.com/DeterminateSystems/flakehub-push/pull/110 because it caused rolling version numbers to be calculated incorrectly.

**Pull request merged**